flask_pydantic/core.py

```diff
@@ -68,30 +103,70 @@ def validate_many_models(model: Type[BaseModel], content: Any) -> List[BaseModel
         raise ManyModelValidationError(ve.errors())
 
 
-def validate_path_params(func: Callable, kwargs: dict) -> Tuple[dict, list]:
-    errors = []
+def validate_path_params(
+    func: ModelRouteCallable, kwargs: Dict[str, Any]
+) -> Tuple[Dict[str, Any], List["ErrorDict"]]:
+    errors: List["ErrorDict"] = []
     validated = {}
     for name, type_ in func.__annotations__.items():
         if name in {"query", "body", "form", "return"}:
             continue
         try:
             value = parse_obj_as(type_, kwargs.get(name))
             validated[name] = value
-        except ValidationError as e:
-            err = e.errors()[0]
-            err["loc"] = [name]
+        except ValidationError as error:
+            err = error.errors()[0]
+            err["loc"] = (name,)
             errors.append(err)
     kwargs = {**kwargs, **validated}
     return kwargs, errors
 
 
-def get_body_dict(**params):
-    data = request.get_json(**params)
+def get_body_dict(**params: Dict[str, Any]) -> Any:
+    data = request.get_json(**params)  # type: ignore
     if data is None and params.get("silent"):
         return {}
     return data
 
 
+def _get_type_generic(hint: Any):
+    """Extract type information from bound TypeVar or Type[TypeVar]."""
+    if isinstance(hint, TypeVar):
+        assert (
+            getattr(hint, "__bound__", None) is not None
+        ), "If using TypeVar, you need to specify bound model."
+        return getattr(hint, "__bound__")
+
+    args = get_args(hint)
+    if len(args) > 0 and isinstance(args[0], TypeVar):
+        assert (
+            getattr(args[0], "__bound__", None) is not None
+        ), "If using TypeVar, you need to specify bound model."
+        return getattr(args[0], "__bound__")
+
+    return hint
```

Comment:
Could you please add tests to cover this? Also would be nice to mention that in docs 🙏

Reply:
Sure thing :slightly_smiling_face:

Reply:
Done. Let me know if it's enough.

Reply:
Hi, any news regarding the review? Is there anything I could do to move forward this PR?